litex/tools/litex_soc_gen.py

```diff
@@ -62,56 +63,66 @@ def build(self, fragment, build_dir, build_name, **kwargs):
 # LiteX SoC Generator ------------------------------------------------------------------------------
 
 class LiteXSoCGenerator(SoCMini):
-    def __init__(self, name="litex_soc", sys_clk_freq=int(50e6), **kwargs):
+    def __init__(self, name="litex_soc", sys_clk_freq=int(50e6), n_mi=3, n_si=0, **kwargs):
+        assert (n_mi + n_si) < 5
+        self.kwargs = kwargs
+
         # Platform ---------------------------------------------------------------------------------
-        platform = Platform(device="", io=get_common_ios())
-        platform.name = name
-        platform.add_extension(get_uart_ios())
+        self.platform = Platform(device="", io=get_common_ios())
```

Comment:
Please don't change code that don't require changes.

Reply:
Hi, I add the "self" to, set it as an attribute as "platform" is needed in the "add_slave_interface" and "slave_master_interface" methods. Sorry for the delay by the way. I was on vacation.